### ISSUE TITLE
Github example : Request not working

### ISSUE THREAD

**Issue by harikt** [auraphp/Aura.Http#22]
``` php
$request = $http->newRequest();
$request->setUrl('https://api.github.com/orgs/auraphp/repos');
$stack = $http->send($request);
$repos = json_decode($stack[0]->content);
foreach ($repos as $repo) {
    echo $repo->name . PHP_EOL;
}
```

``` sh
PHP Fatal error:  Uncaught exception 'OutOfRangeException' with message 'Offset invalid or out of range' in /var/www/contributors/example.php:6
Stack trace:
#0 /var/www/contributors/example.php(6): SplDoublyLinkedList->offsetGet(0)
#1 {main}
  thrown in /var/www/contributors/example.php on line 6
```

**Comment by harikt** [comment]
I figured out the issue. 

But I assume this also need to be fixed at the code level. Problem is even if the response is given by github it is not set to the stack in StackBuilder . 

It is not entering to here https://github.com/auraphp/Aura.Http/blob/develop/src/Aura/Http/Message/Response/StackBuilder.php#L80 .

The reason was there was no user agent set at our end if the user has not set in headers.

**Comment by pmjones** [comment]
Fixed by #24
